### PR TITLE
Document the ability to deserialize into JsonElement properties

### LINKED ISSUE #24501: Document the ability to deserialize into JsonElement properties
Somewhat related to overflow deserialization, we can also do this (using a similar example as the [overflow page](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-handle-overflow)):

```cs
public class WeatherForecast
{
    public DateTimeOffset Date { get; set; }
    public int TemperatureCelsius { get; set; }
    public string Summary { get; set; }
    public JsonElement DatesAvailable { get; set; }    // <= serialize generic JSON array
    public JsonElement SummaryWords { get; set; }  // <= serialize generic JSON element
}
```

```json
{
  "Date": "2019-08-01T00:00:00-07:00",
  "TemperatureCelsius": 25,
  "Summary": "Hot",
  "DatesAvailable": [
    "2019-08-01T00:00:00-07:00",
    "2019-08-02T00:00:00-07:00"
  ],
  "SummaryWords": {
    "Cool": "Stuff",
    "Windy": "Wind",
    "Humid": "Water"
  }
}
```

I couldn't find this capability documented anywhere.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 588ae32f-3429-143f-824c-5ab2d046d24a
* Version Independent ID: 002f9f2a-48d1-0671-a079-b99b9f6bab58
* Content: [How to serialize and deserialize JSON using C# - .NET](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-how-to?pivots=dotnet-5-0#how-to-read-json-as-net-objects-deserialize)
* Content Source: [docs/standard/serialization/system-text-json-how-to.md](https://github.com/dotnet/docs/blob/main/docs/standard/serialization/system-text-json-how-to.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #24501 

[Internal review URL](https://review.docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-handle-overflow?branch=pr-en-us-24522)

### CODE REVIEW

File: docs/standard/serialization/system-text-json-handle-overflow.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ ms.topic: how-to
 
 # How to handle overflow JSON with System.Text.Json
 
-In this article, you will learn how to handle overflow JSON with the `System.Text.Json` namespace.
+This article shows how to handle overflow JSON with the `System.Text.Json` namespace.
```
Comment (luisquintanilla) [inline docs/standard/serialization/system-text-json-handle-overflow.md:19]: Consider adding an xref 

```suggestion
This article shows how to handle overflow JSON with the [`System.Text.Json`](xref:System.Text.Json) namespace.
```

**Review by luisquintanilla — APPROVED** [review]
@tdykstra looks great. Added a minor suggestion

File: docs/standard/serialization/system-text-json-handle-overflow.md [hunk 2]
```diff
@@ -49,16 +49,16 @@ If you deserialize the JSON shown into the type shown, the `DatesAvailable` and
...
 | Property | Value | Notes |
 |--|--|--|
 | `Date` | `"8/1/2019 12:00:00 AM -07:00"` |  |
 | `TemperatureCelsius` | `0` | Case-sensitive mismatch (`temperatureCelsius` in the JSON), so the property isn't set. |
 | `Summary` | `"Hot"` |  |
 | `ExtensionData` | `temperatureCelsius: 25` | Since the case didn't match, this JSON property is an extra and becomes a key-value pair in the dictionary. |
-| `DatesAvailable` | `[ "8/1/2019 12:00:00 AM -07:00", "8/2/2019 12:00:00 AM -07:00" ]` | Extra property from the JSON becomes a key-value pair, with an array as the value object. |
-| `SummaryWords` | `[ "Cool", "Windy", "Humid" ]` | Extra property from the JSON becomes a key-value pair, with an array as the value object. |
+| `ExtensionData` | `DatesAvailable: [ "8/1/2019 12:00:00 AM -07:00", "8/2/2019 12:00:00 AM -07:00" ]` | Extra property from the JSON becomes a key-value pair, with an array as the value object. |
+| `ExtensionData` | `SummaryWords: [ "Cool", "Windy", "Humid" ]` | Extra property from the JSON becomes a key-value pair, with an array as the value object. |
```
Comment (layomia) [inline docs/standard/serialization/system-text-json-handle-overflow.md:61]: Should these be collapsed into a single `ExtensionData` row, and the value displayed with JSON-object representation?

File: docs/standard/serialization/system-text-json-handle-overflow.md [hunk 3]
```diff
@@ -82,6 +82,16 @@ When the target object is serialized, the extension data key value pairs become
 
 Notice that the `ExtensionData` property name doesn't appear in the JSON. This behavior lets the JSON make a round trip without losing any extra data that otherwise wouldn't be deserialized.
 
+The following example shows a round trip from JSON to a deserialized object and back to JSON:
+
+:::code language="csharp" source="snippets/system-text-json-how-to/csharp/RoundtripExtensionData.cs" highlight="13-14":::
+
+## Deserialize into JsonElement
+
+If you just want to be flexible about what JSON to accept for a particular property, an alternative is to deserialize into <xref:System.Text.Json.JsonElement>. Any valid JSON property can be deserialized into `JsonElement`. The following example shows a round trip from JSON and back to JSON for a class that includes properties of type `JsonElement`.
```
Comment (layomia) [inline docs/standard/serialization/system-text-json-handle-overflow.md:91]: Should we create a .NET 6.0 partition and mention the `JsonNode` type? `JsonElement` would be used if immutability is okay, whereas `JsonNode` offers mutability.

**Review by layomia — APPROVED** [review]
Some minor questions, but LGTM.

File: docs/standard/serialization/system-text-json-handle-overflow.md (hunk 3 above)
Comment (tdykstra) [reply docs/standard/serialization/system-text-json-handle-overflow.md:91]: I'll create a new issue for this and include it in the PR that creates the .NET 6 Preview zone pivot.

**Pull request merged**